### DISCUSSION TITLE
esp32 img tag

### DISCUSSION THREAD

**Discussion by Senso4sLab** [Mair/esp32-course#118] [Course Qeustions] [answered]
I have a website on which the img tag is added. The entire website is stored in the custom partition named "stiff". The request for loading image is executed properly but image is not shown (just white rectangle). Here is my code:

    ```
     FILE *file = fopen(path, "r");     
     if(file == NULL)     
     {
        httpd_resp_send_404(r);
        esp_vfs_spiffs_unregister(NULL);     
        return ESP_OK;
     }   
     
     char lineRead[256];
     while(fgets(lineRead,sizeof(lineRead), file))     
     {
        httpd_resp_sendstr_chunk(r,lineRead);
     } 
     fclose(file);     
     httpd_resp_sendstr_chunk(r,NULL);
     esp_vfs_spiffs_unregister(NULL);    
```
It is worth mentioning that the path is correct because the variable file is not equal to zero. I was trying to add EMBED_FILES in spiffs_create_partition_image, such as 
```
spiffs_create_partition_image(spiffs ../webpage FLASH_IN_PROJECT  INCLUDE_DIRS "." EMBED_FILES /images/Ball.jpg)
```

and then try to call the image based on the following variables

```
extern const unsigned char ball_start[] asm("_binary_Ball_jpg_start");
extern const unsigned char ball_end[] asm("_binary_Ball_jpg_end");
```

but the problem is that this approach does not even compile. I do not have any other idea about how to solve this problem.  It is interesting, if I add EMBED_FILES in idf_component_register then everything works just fine. For example:

```
idf_component_register(SRCS "station_example_main.c"
                    INCLUDE_DIRS "."
                    EMBED_FILES  ../webpage/images/Ball.jpg 
                    )
```

but in this case, the image is loaded from the app partition and not from a custom one. My question is how can I load images from a custom partition?

Thanks in advance.

**Comment by Mair** [comment, marked as answer]
you need to set the correct mime type for images
https://github.com/Mair/esp32-course/blob/8dd1dd11ec1720197f93048a18533cdd515f8325/_15_Internet_Server/v14/_11_web_site_10_change_networks/main/main.c#L51
